### ISSUE TITLE
[Bug]: Accessibility snapshot does not include iframe contents

### ISSUE THREAD

**Issue by andyearnshaw** [puppeteer/puppeteer#7744]
### Bug description

Steps to reproduce the problem:

1. Create a page with an iframe, in the iframe place a page with a button
2. Call `console.log(await page.accessibility.snapshot())`
3. Observe that the snapshot does not include the iframe contents on the `role: 'Iframe'` node
4. Execute the following:
```javascript
const button = await page.mainFrame().childFrames()[0].$('button');
console.log(await page.accessibility.snapshot({ root: button });
```
5. Observe that the console log result is `null`

Here's a sample output from my app:

```javascript
{
  role: 'WebArea',
  name: 'Empty Page',
  children: [
    { role: 'link', name: 'Apps' },
    { role: 'heading', name: 'Site navigation', level: 2 },
    { role: 'menubar', name: 'Site navigation', children: [Array] },
    { role: 'heading', name: 'Quick navigation', level: 2 },
    { role: 'button', name: 'Search', pressed: false },
    {
      role: 'menuitem',
      name: '2 unread notifications Show notifications menu',
      haspopup: 'menu'
    },
    { role: 'menuitem', name: 'Help menu', haspopup: 'menu' },
    { role: 'menuitem', name: 'Profile menu', haspopup: 'menu' },
    { role: 'Iframe', name: 'Content' }
  ]
}
```

Note that the last item with `role: 'Iframe'` has no `children` array. This does not change if you add the `interestingOnly: false` option. Additionally, if you try to call `snapshot()` with `root` set to any element handle inside the iframe, it will return `null`. This means, unless I'm missing something in the API, it's not possible to get an accessibility snapshot of any iframe, as `accessibility` does not exist on the `Frame` class. However, it is possible to use ARIA queries to find things inside the iframe, as long as the query is performed from one of the target's ancestors up to and including the iframe handle (which also feels wrong to me that I can't query any AX node from the top frame).

It looks like there are two problems [here](https://github.com/puppeteer/puppeteer/blob/f863f4bfe015e57ea1f9fbb322f1cedee468b857/src/common/Accessibility.ts#L185-L200). One is that the result of CDP's `Accessibility.getFullAXTree` does not contain iframe content and the other is that, when `root` is specified, Puppeteer still calls `Accessibility.getFullAXTree` instead of `Accessibility.getPartialAXTree`.

If I copy the accessibility tree from `chrome://accessibility`, the full tree is returned. If I inspect the accessibility tree for an element in the dev tools there is no distinction of an iframe or page boundary, the hierarchy goes all the way up to the main frame root. My understanding is that the accessibility tree is a merged tree containing all nodes for the current window, so I think the fix should be to expand iframe nodes by either adding an option to merge trees upstream at `Accessibility.getFullAXTree` or by recursing the nodes and calling `Accessibility.getFullAXTree` again on any iframe nodes. Additionally, calling `Accessibility.getPartialAXTree` when a `root` is specified might be appropriate to improve performance.

### Puppeteer version

v10.2.0

### Node.js version

v16.12.0

### npm version

v8.1.0

### What operating system are you seeing the problem on?

macOS

**Comment by andyearnshaw** [comment]
Is there anything I can do to help confirm this?

**Comment by OrKoN** [comment]
@andyearnshaw sorry about it, we tried to deal with thousands of bugs (often without repro) that accumulated over the years using the stale bot. Please ping me on the issues that are still relevant. Re-submitting the issue using the latest version would also make it pop-up in our queues.

**Comment by crunchy-vonage** [comment]
Same issue
114.0.5735.198 (Official Build) (x86_64)
The chrome built in tools do display the iframes, getFullAXTree does not
My preference would be an API that behaves the same as UI version in chrome.

**Comment by ElliottStorey** [comment]
Same issue here using the latest version (23.10.1). Any progress?
